drivers/serial/uart_stm32.c

```diff
@@ -1257,7 +1257,6 @@ static void uart_stm32_isr(const struct device *dev)
 			LL_USART_IsActiveFlag_TC(config->usart)) {
 
 		LL_USART_DisableIT_TC(config->usart);
-		LL_USART_ClearFlag_TC(config->usart);
```

Comment:
Did you test this with with PM_DEVICE enabled ?
At least, one issue I can see is that you'll likely be blocked by this piece of code in `uart_stm32_suspend_setup`
https://github.com/zephyrproject-rtos/zephyr/blob/a51bd53cef2b2b6cacb1dc7a366adb45565659f3/drivers/serial/uart_stm32.c#L2054-L2055

At the minimum, it would have to be changed as well.
Did you test your code against PM+DMA samples ?

Reply:
Yes, I'm using `PM=y`, `PM_DEVICE=y` and async API which means DMA.

I'm blocked **without** this PR. This PR removes clearing TC flag.
```
 while (LL_USART_IsActiveFlag_TC(config->usart) == 0) { 
 } 
```
this code waits for `TC=1`. With clearing TC flag in ISR, right after the transfer ends, it waits forever.

Reply:
> this code waits for TC=1. With clearing TC flag in ISR, right after the transfer ends, it waits forever.

Of course...
Coffee time !